litestar/_openapi/path_item.py

```diff
@@ -1,141 +1,137 @@
 
-def extract_layered_values(
-    route_handler: HTTPRouteHandler,
-) -> tuple[list[str] | None, list[dict[str, list[str]]] | None]:
-    """Extract the tags and security values from the route handler layers.
+            if not route_handler.resolve_include_in_schema():
+                continue
 
-    Args:
-        route_handler: A Route Handler instance.
+            operation = self.create_operation_for_handler_method(route_handler, HttpMethod(http_method))
 
-    Returns:
-        A tuple of optional lists.
-    """
-    tags: list[str] = []
-    security: list[SecurityRequirement] = []
-    for layer in route_handler.ownership_layers:
-        if layer.tags:
-            tags.extend(layer.tags)
-        if layer.security:
-            security.extend(layer.security)
-    return sorted(set(tags)) if tags else None, security or None
-
-
-def create_path_item(
-    create_examples: bool,
-    operation_id_creator: OperationIDCreator,
-    plugins: Iterable[OpenAPISchemaPluginProtocol],
-    route: HTTPRoute,
-    schemas: dict[str, Schema],
-    use_handler_docstrings: bool,
-) -> tuple[PathItem, list[str]]:
+            setattr(self._path_item, http_method.lower(), operation)
+
+        return self._path_item
+
+    def create_operation_for_handler_method(
+        self, route_handler: HTTPRouteHandler, http_method: HttpMethod
+    ) -> Operation:
+        """Create an Operation instance for a given route handler and http method.
+
+        Args:
+            route_handler: A route handler instance.
+            http_method: An HttpMethod enum value.
+
+        Returns:
+            An Operation instance.
+        """
+        operation_id = self.create_operation_id(route_handler, http_method)
+        parameters = create_parameters_for_handler(self.context, route_handler, self.route.path_parameters)
+        signature_fields = route_handler.parsed_fn_signature.parameters
+
+        request_body = None
+        if data_field := signature_fields.get("data"):
+            request_body = create_request_body(
+                self.context, route_handler.handler_id, route_handler.resolve_data_dto(), data_field
+            )
+
+        raises_validation_error = bool("data" in signature_fields or self._path_item.parameters or parameters)
+        responses = create_responses_for_handler(
+            self.context, route_handler, raises_validation_error=raises_validation_error
+        )
+
+        return route_handler.operation_class(
+            operation_id=operation_id,
+            tags=route_handler.resolve_tags() or None,
+            summary=route_handler.summary or SEPARATORS_CLEANUP_PATTERN.sub("", route_handler.handler_name.title()),
+            description=self.get_description_for_handler(route_handler),
+            deprecated=route_handler.deprecated,
+            responses=responses,
+            request_body=request_body,
+            parameters=parameters or None,  # type: ignore[arg-type]
```

Comment:
Is this type ignore needed because `mypy` doesn't like `list[Parameter]` for `list[Parameter | Reference]`?

Reply:
I haven't looked into it actually, we have the same in main: https://github.com/litestar-org/litestar/blob/7979771d941ab2cc3be6249f59d31dc48ccc6039/litestar/_openapi/path_item.py#L135